server/player/player.go

```diff
@@ -670,13 +641,25 @@ func (d *Data) GiveItem(newItem item.Data, db *sql.DB) error { // TODO: Refactor
 }
 
 // TakeItem from Data
-func (d *Data) TakeItem(id int32, slot int16, amount int16, invID byte) {
-
+func (d *Data) TakeItem(id int32, slot int16, amount int16, invID byte, db *sql.DB) item.Data {
 	item, err := d.getItem(invID, slot)
 	if err != nil {
-		return
+		log.Println(err)
 	}
-	d.Send(packetInventoryRemoveItem(item))
+
+	maxRemove := math.Min(float64(item.Amount()), float64(amount))
+
+	item.UpdateAmount(item.Amount() - int16(maxRemove))
+
+	if item.Amount() == 0 {
+		// Delete item
+		d.removeItem(item, db)
+	} else {
+		//Update item with new amount
```

Comment:
`packetInventoryAddItem` will probably do what yo want here.

Reply:
That packet works great. I need to implement the DB side of things and then I'll push that commit into this PR.

Reply:
I am not sure why but the update is not persisting past a single use. For example if you use 1 potion it updates fine, but if you use 3 in a row it will still act as if it is one use. So rather than 10 -> 7 for 3 potions in a row it will be 10 -> 9. The db is updating after usage as well, but because the item in memory is not updating past the single use the DB is not getting updated correctly either. 

I have reviewed the code and I am not seeing where this issue could be occurring at. Do you have any insight on this?

Reply:
`getItem` is returning a copy of the item, you need a ptr.

Reply:
I was able to get it working by using the item Save and updateItem functions.